lsp-fiber/src/rpc.ml

```diff
@@ -277,9 +277,10 @@ struct
       Fiber.Cancel.with_handler cancel
         ~on_cancel:(fun () -> on_cancel jsonrpc_req.id)
         (fun () ->
-          let+ resp =
-            Session.request_with_cancel (Fdecl.get t.session) cancel jsonrpc_req
+          let _, req_f =
```

Comment:
I'm not sure if we need to use and fire the `cancel` in the `on_cancel`? The `lsp-fiber` cancellation test works in both cases..

Reply:
Using the `cancel` will allow the request to proceed without waiting for a response from the server. So it's a matter if want to trust the server to send the RequestCancelled response or we just ignore it and move on. I elected to go for the conservative route here for now and just wait for the cancellation response.